### PR TITLE
Add pre-install checks for required python packages

### PR DESCRIPTION
**Pull request by sachinpunadikar**
"cortx-py-utils" rpm depends on various python packages.
The %pre phase should check whether these python packages are
installed or not. If these packages do not exists, then throw
message asking to install them

-----
[View rendered py-utils/README.md](https://github.com/sachinpunadikar/cortx-utils/blob/pre_install_checks/py-utils/README.md)

### CODE REVIEW

**Review by rahul-tripathi-git — APPROVED** [review]
Looks Good to me.

**Comment by sachinpunadikar** [comment]
Please ignore the mini-provisioner failure. Earlier job went well
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-utils/84/console

File: py-utils/utils-pre-install [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+# Remember to sync below list with "requirements.txt" file.
+
+PACKAGE_LIST="aiohttp==3.6.1
```
Comment (ujjwalpl) [inline py-utils/utils-pre-install:19]: Can we generate this from requiremets.txt during build time to avoid possibility of manual error.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/utils-pre-install (hunk 1 above)
Comment (sachinpunadikar) [reply py-utils/utils-pre-install:19]: @ujjwalpl, I was not able to locate any method, which will help in updating/generating a file when we run setup.py bdist_rpm. Let me know if you have any idea on how to do that.

Comment (sachinpunadikar) [reply py-utils/utils-pre-install]: @ujjwalpl as suggested, created a wrapper script for building cortx-py-utils. This takes care of generating "utils-pre-install".

**Review by rahul-tripathi-git — APPROVED** [review]

File: py-utils/rpm_build.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
```
Comment (ujjwalpl) [inline py-utils/rpm_build.sh:1]: Lets keep this to the following location to be consistent. 
<repo root>/jenkins/build.sh

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/rpm_build.sh (hunk 2 above)
Comment (sachinpunadikar) [reply py-utils/rpm_build.sh]: Done

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**